data/trainers.h and data/battle_partners.h not generating after make clean && make

### Description

Upon a fresh build, data/trainers.h and data/battle_partners.h generate but after a `make clean`, as per usual the files are deleted but they don't regenerate upon the next `make` causing the following errors:

```/usr/lib/gcc/arm-none-eabi/10.3.1/cc1 -quiet <flags> -o build/modern/src/battle_tower.o src/battle_tower.c
src/battle_tower.c:703:10: fatal error: data/battle_partners.h: No such file or directory
  703 | #include "data/battle_partners.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
src/battle_tower.c:702:1: error: expected expression at end of input
  702 | {
      | ^
src/battle_tower.c:51:13: warning: 'InitTowerChallenge' declared 'static' but never defined [-Wunused-function]
   51 | static void InitTowerChallenge(void);
```
Other similar errors follow this one but clearly caused by those files not regenerating after `make clean` and this is also happening on the upcoming branch as of the latest commit 7f849ef

### Version

master (default, unreleased bugfixes)

### Upcoming/master Version

be03fec

### Discord contact info

pk87

--- a/Makefile
+++ b/Makefile
@@ -350,9 +350,6 @@ clean-generated:
 COMPETITIVE_PARTY_SYNTAX := $(shell PATH="$(PATH)"; echo 'COMPETITIVE_PARTY_SYNTAX' | $(CPP) $(CPPFLAGS) -imacros include/gba/defines.h -imacros include/config/general.h | tail -n1)
 ifeq ($(COMPETITIVE_PARTY_SYNTAX),1)
 %.h: %.party ; $(CPP) $(CPPFLAGS) -traditional-cpp - < $< | $(TRAINERPROC) -o $@ -i $< -
-
-AUTO_GEN_TARGETS += $(DATA_SRC_SUBDIR)/trainers.h
-AUTO_GEN_TARGETS += $(DATA_SRC_SUBDIR)/battle_partners.h
 endif
 
 $(C_BUILDDIR)/librfu_intr.o: CFLAGS := -mthumb-interwork -O2 -mabi=apcs-gnu -mtune=arm7tdmi -march=armv4t -fno-toplevel-reorder -Wno-pointer-to-int-cast

